src/backend/distributed/utils/maintenanced.c

```diff
@@ -158,7 +184,7 @@ InitializeMaintenanceDaemonBackend(void)
 		memcpy(worker.bgw_extra, &extensionOwner, sizeof(Oid));
 		worker.bgw_notify_pid = MyProcPid;
 
-		if (!RegisterDynamicBackgroundWorker(&worker, &handle))
+		if (!RegisterDynamicBackgroundWorker(&worker, &dbData->workerHandle))
```

Comment:
That can't work - the pointer is set to process local memory. So other processes can't access it after this.